cob_twist_controller/include/cob_twist_controller/cob_twist_controller.h

```diff
@@ -147,8 +153,21 @@ class CobTwistController
 	void twist_stamped_cb(const geometry_msgs::TwistStamped::ConstPtr& msg);
 	void solve_twist(KDL::Twist twist);
 	
-	KDL::JntArray normalize_velocities(KDL::JntArray q_dot_ik);
-	
+	/**
+	 * Needed for limiting the base velocities
+	 * @param The inverse kinematics solution for the joint velocity vector
+	 * @return The normalized joint velocity vector
+	 */
+	KDL::JntArray normalize_velocities(const KDL::JntArray &q_dot_ik) const;
+	
+	/**
+     * Check whether joint velocities exceed limits or not.
+	 * @param last_q The latest given joint positions.
+	 * @param new_q_dot_ik The new calculated joint velocities.
+	 * @return Joint velocities adapted to limits if exceeded.
+	 */
+	KDL::JntArray enforce_limits(const KDL::JntArray& q, const KDL::JntArray& q_dot_ik) const;
```

Comment:
See latest latest commit 5703cdc9cf (20th April '15). 
New structure for enforcing limits -> new classes which can be concatenated and made independent. 

Added keep_direction parameter to cfg to choose between all and individual.